stixcore/io/fits/processors.py

```diff
@@ -0,0 +1,140 @@
+import logging
+from datetime import datetime
+
+import numpy as np
+
+from astropy.io import fits
+
+from stixcore.util.logging import get_logger
+
+__all__ = ['SEC_IN_DAY', 'FitsLBProcessor']
+
+logger = get_logger(__name__, level=logging.DEBUG)
+
+
+SEC_IN_DAY = 24 * 60 * 60
+
+
+class FitsProcessor:
+    # abstract some general processing pattern methods
+    pass
+
+
+class FitsLBProcessor(FitsProcessor):
+    def __init__(self, archive_path):
+        self.archive_path = archive_path
+
+    def write_fits(self, product):
+
+        for prod in product.to_days():
+            filename = self.generate_filename(prod, version=1)
+            if getattr(prod, 'ssid', False):
```

Comment:
Not sure if it will ever have `ssid` with the current code?

Reply:
fixed type now also includes a ssis/sp1_val